legacy/builder/resolve_library.go

```diff
@@ -48,7 +48,16 @@ func ResolveLibrary(ctx *types.Context, header string) *libraries.Library {
 
 	selected := resolver.ResolveFor(header, ctx.TargetPlatform.Platform.Architecture)
 	if alreadyImported := importedLibraries.FindByName(selected.Name); alreadyImported != nil {
-		selected = alreadyImported
+		// Certain libraries might have the same name but be different.
+		// This usually happens when the user includes two or more custom libraries that have
+		// different header name but are stored in a parent folder with identical name, like
+		// ./Lib1/src/lib1.h and ./Lib2/src/lib2.h
```

Comment:
```suggestion
		// ./Libraries1/Foo/lib1.h and ./Libraries2/Foo/lib2.h
```
The previous example is confusing because typically when you see a `src` folder it implies an Arduino library with the "recursive" layout, like this:
```
Lib1
|_ library.properties
|_ src
   |_ lib1.h
```
The correct usage of the `--libraries` flag is to pass a path to a directory containing libraries (which would be the parent of the `Lib1` folder with this structure, not the path of a library (which would be the `Lib1` folder in this structure. Since the example implies the latter, which would not trigger the bug since `Lib1` != `Lib2`, it might be confusing to future readers.

After downloading the TestProject.zip attachment from https://github.com/arduino/arduino-cli/issues/973, I now understand that the `src` folders actually contain legacy libraries, but the future reader of this comment should not need that context.